test/dotnet-test.Tests/AssemblyInfo.cs

```diff
@@ -0,0 +1,4 @@
+using Xunit;
+
+// Don't let the tests execute concurrently
+[assembly: CollectionBehavior(CollectionBehavior.CollectionPerAssembly)]
```

Comment:
@pranavkm Make sure that there aren't multiple tests with the same name. I mean, test methods with the same name in different classes. If there are, you may hit this.